### PR TITLE
[math] Detach TEventList instances from gDirectory

### PR DESCRIPTION
**Pull request by dpiparo**
fixes ROOT-8501
https://its.cern.ch/jira/browse/ROOT-8501

### CODE REVIEW

File: math/mlp/src/TMultiLayerPerceptron.cxx [hunk 1]
```diff
@@ -458,9 +466,15 @@ TMultiLayerPerceptron::TMultiLayerPerceptron(const char * layout, TTree * data,
    fData = data;
    fCurrentTree = -1;
    fCurrentTreeWeight = 1;
-   fTraining = new TEventList(Form("fTrainingList_%zu",(size_t)this));
+   {
+      RGDirectoryDetacher raii;
```
Comment (vepadulano) [inline math/mlp/src/TMultiLayerPerceptron.cxx:470]: I believe this should be done via
```suggestion
      TDirectory::TContext ctxt;
```

**Review by vepadulano — CHANGES_REQUESTED** [review]
Nice! But see comment

File: math/mlp/src/TMultiLayerPerceptron.cxx (hunk 1 above)
Comment (dpiparo) [reply math/mlp/src/TMultiLayerPerceptron.cxx:470]: absolutely. Sorry for forgetting about that.

**Comment by dpiparo** [comment]
@lmoneta it seems all comments have been addressed (sorry for the initial sloppy implementation): do you think we can merge and close the issue?

**Review by lmoneta — APPROVED** [review]
LGTM!
Tahnk you Danilo for fixing this issue!

**Pull request merged**